File: C++/A/984A-game.cpp
#include <bits/stdc++.h>
using namespace std;

class Solver
{
    int x;
    vector<int> numbers;

public:
    void input()
    {
        cin >> x;
        for (int i = 0; i < x; i++)
        {
            int a;
            cin >> a;
            numbers.push_back(a);
        }
    }
    int answer()
    {
        sort(numbers.begin(), numbers.end());
        return numbers[(x - 1) / 2];
    }
};

int main()
{
    Solver solution;
    solution.input();
    int y = solution.answer();
    cout << y;
}

File: Python/984A-game.py
class Solver:
    def __init__(self):
        self.x = 0
        self.numbers = []

    def input(self):
        self.x = int(input())
        self.numbers = list(map(int, input().split()))

    def answer(self):
        self.numbers = sorted(self.numbers)
        return self.numbers[(self.x-1)//2]


solution = Solver()
solution.input()
print(solution.answer())
